text/2996-async-iterator.md

```diff
@@ -149,33 +150,33 @@ impl Stream for Counter {
 There are a number of simple "bridge" impls that are also provided:
 
 ```rust
-impl<S> Stream for Box<S>
+impl<S> AsyncIterator for Box<S>
 where
-    S: Stream + Unpin + ?Sized,
+    S: AsyncIterator + Unpin + ?Sized,
 {
-    type Item = <S as Stream>::Item
+    type Item = <S as AsyncIterator>::Item
 }
 
-impl<S> Stream for &mut S
+impl<S> AsyncIterator for &mut S
 where
-    S: Stream + Unpin + ?Sized,
+    S: AsyncIterator + Unpin + ?Sized,
 {
-    type Item = <S as Stream>::Item;
+    type Item = <S as AsyncIterator>::Item;
 }
 
-impl<S, T> Stream for Pin<P>
+impl<S, T> AsyncIterator for Pin<P>
 where
     P: DerefMut<Target=T> + Unpin,
```

Comment:
I think the `P: Unpin` bound is unnecessary. See https://github.com/rust-lang/rust/pull/81363.

Reply:
You may be right, but the `P: Unpin` bound [exists on the implementation today](https://doc.rust-lang.org/std/stream/trait.Stream.html#impl-Stream-2), and is separate from the change we're proposing here. Likely the best way to go about this would be to open a separate PR against the stdlib, reasoning why you think we can drop the `P: Unpin` bound for `Stream`.